src/bin/cargo/commands/install.rs

```diff
@@ -119,6 +120,7 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         .get_many::<CrateVersion>("crate")
         .unwrap_or_default()
         .cloned()
+        .dedup_by(|x, y| x.0 == y.0)
```

Comment:
Should we pick up a bigger version here? For example: cargo install foo@1 foo@2 Should we pick the last one here?

Reply:
What about only de-duping when they are going to install the same version, and giving an error (or warning?) when installing different versions of the same package?

Reply:
Alternatively, don't de-dup. We only emit warnings at this time, so that we can avoid breaking stuff if people really do odd things like <https://github.com/rust-lang/cargo/issues/12866#issuecomment-1774101768> mentioned.

Reply:
FYI I think we need some more info from the author before we act on this.

Reply:
> when installing different versions of the same package?

It seems we don't have the ability to support different versions. So always pick the newer one is better?

> We only emit warnings at this time, so that we can avoid breaking stuff if people really do odd things like [#12866 (comment)](https://github.com/rust-lang/cargo/issues/12866#issuecomment-1774101768) mentioned.

This behavior is too odd. At least, for the same version, I don't think we break anything because we always get the same binary installed.

Reply:
Say, running `cargo install mycompiler@2.0.0 mycompiler@3.0.0`. People rely on the old version to install a new version, done in their `build.rs` as a bootstrap process. This is rare but technically possible.

It could start as a warning. Once the warning hits stable, the transition period ends and turns it into picking the newer one as proposed.

What do you think? It might be I am overthinking.

Reply:
For myself, that is a corner case that I don't expect to be an issue.  However, having gotten more information from the author, I feel like the value gained by this change is below the cost of these conversations.  For myself, I'd say if there is little discuss and we can merge it, fine, otherwise close it.

Reply:
Fine with either as well. If we want to merge it, yes taking the larger version sounds good to me.

Reply:
Fine with either as well. If we want to merge it, yes taking the last one sounds good to me.

Reply:
@epage @weihanglo I updated it to only dedup the same version. I think it is ready to go. We don't really break anything for this situation.